src/stack.h

```diff
@@ -39,10 +39,31 @@ typedef struct HSSlice {
     } data;
 } HSSlice;
 
-typedef struct HSStack {
+class HSStack {
+public:
+    HSStack() = default;
+    ~HSStack();
+
+    void insert_slice(HSSlice* elem);
+    void remove_slice(HSSlice* elem);
+    void raise_slide(HSSlice* slice);
+    void mark_dirty();
+    void slice_add_layer(HSSlice* slice, HSLayer layer);
+    void slice_remove_layer(HSSlice* slice, HSLayer layer);
+    bool is_layer_empty(HSLayer layer);
+    void clear_layer(HSLayer layer);
+
+    // returns the number of windows in this stack
+    int window_count(bool real_clients);
+    void to_window_buf(Window* buf, int len, bool real_clients, int* remain_len);
+    void restack();
+    Window lowest_window();
+
     GList*  top[LAYER_COUNT];
```

Comment:
While we are at it, could we also get rid of this?
How is this used? Like an std::list? Or even std::forward_list?

Reply:
It's simply a `vector<HSSlice*> top[LAYER_COUNT]`.